### PR TITLE
Prevent User Relocation of Fan Subnets

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This patch prevents fan overlay subnets from being moved via the `MoveSubnets` method on the spaces API facade. The error message returned from such an attempt advises moving the underlay instead.

Included are mechanical testing changes:
- Mocks are moved into the `spaces` package and relocated to _package_mock_test.go_.
- _package_test.go_ is now local to the `spaces` package and exports a base suite for tests in `spaces_test` to use.
- The _export_test.go_ file is deleted and anything still required moved to _package_test.go_.

Some further modularisation of the test suites should be made, but in the interests of brevity that is eschewed here.

## QA steps

QA steps to follow with the CLI implementation. Unit tests cover this logic.

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Spaces API - disallow moving Fan overlays.](https://trello.com/c/66IRLPCd/2436-spaces-api-disallow-moving-fan-overlays)

File: apiserver/facades/client/spaces/move.go [hunk 1]
```diff
@@ -127,7 +129,7 @@ func (api *API) MoveSubnets(args params.MoveSubnetsParams) (params.MoveSubnetsRe
 			continue
 		}
 
-		if err := api.ensureSubnetCanBeMoved(subnets, spaceName, toSpaceParams.Force); err != nil {
+		if err := api.ensureSubnetsCanBeMoved(subnets, spaceName, toSpaceParams.Force); err != nil {
```
Comment (SimonRichardson) [inline apiserver/facades/client/spaces/move.go]: Great minds, this was on in my PR locally, I'll update my local one :)

File: apiserver/facades/client/spaces/move.go [hunk 2]
```diff
@@ -173,7 +175,14 @@ func (api *API) getMovingSubnets(tags []string) ([]MovingSubnet, error) {
 // based on application space constraints.
 // We need to handle the scenario where applications are *bound* to spaces
 // that would mutate as a result of moving subnets.
-func (api *API) ensureSubnetCanBeMoved(subnets []MovingSubnet, spaceName string, force bool) error {
+func (api *API) ensureSubnetsCanBeMoved(subnets []MovingSubnet, spaceName string, force bool) error {
+	for _, subnet := range subnets {
+		if subnet.FanLocalUnderlay() != "" {
```
Comment (SimonRichardson) [inline apiserver/facades/client/spaces/move.go]: Nice :+1:

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**